### DISCUSSION TITLE
Add a stroke effect around image

### DISCUSSION THREAD

**Discussion by JanCourtheyn** [ImageMagick/ImageMagick#3575] [Help]
Hi everyone,

I'm working on a specific usecase via Imagemagick.
I have included the original image, the photoshop result and the result I achieved this far with Imagemagick.

**The original file**
![input](https://user-images.githubusercontent.com/83012886/115709583-aa604580-a371-11eb-957d-99d91f8d8685.png)
**The photoshop result**
![photoshop](https://user-images.githubusercontent.com/83012886/115710510-b8fb2c80-a372-11eb-9e9d-82a9537b0b47.png)
The manipulations that are done is:

- Resize the image to 290x 
- Rotate the image
- Set Threshold to 15%
- Set colorspace to grey
- Stroke the image by 10px border
- Increase canvas to: 200x300
- Set DPI to 200

**The imagemagick result**
![imagemagick](https://user-images.githubusercontent.com/83012886/115709578-a9c7af00-a371-11eb-85ba-341cd49b954b.jpg)
This is the code line i use:
`magick.exe "%%INPUTfile%%"[4] -resize 290x -rotate 90 -gravity center -extent 200x300 -colorspace gray -threshold 15% -density 200x200 "%%OUTPUTfile%%"`

The reason for the [4] is because it is a PSD file the original one where I select layer 4.

So I got quiet close to the photoshop result, but I would need the stroke effect, any tips on how to achieve this?

Thanks in advance!
Jan

**Comment by snibgo** [comment]
You don't link to your input image. If it has a transparent background, "-alpha extract" will make an image that is black where it was transparent, and white where it was opaque. "-edge" will find the edge, and and be used with "-compose Darken -composite" to paint the outline.

If you don't have transparency, you can turn non-white pixels black with "-fill Black +opaque White" and continue with "-edge".

**Reply by JanCourtheyn** [reply]
Thanks for the response.

It has a transparent background since it's a PSD file.
And not sure if you completly got the request, the request would be to outline the original image, like in the photoshop file, the outer edge of the screw should be stroked black.  

![image](https://user-images.githubusercontent.com/83012886/115725596-48f4a280-a382-11eb-9f98-b6945ec0ac4b.png)

Could you give me a hand on how the line would look then? Just tried this but it does nothing:
`magick.exe "inputimage"[4] -alpha extract -edge -compose Darken -composite -resize 290x -rotate 90 -gravity center -extent 200x300 -colorspace gray -threshold 15% -density 200x200 "outputimage"
`

Link to PSD file:
https://we.tl/t-FaRJ5N6fq7

**Comment by snibgo** [comment]
Please link to your input PSD file.

**Reply by JanCourtheyn** [reply]
See update on previous comment. Thanks.

**Comment by snibgo** [comment]
wetransfer.com needs me to accept their Terms of Service, which I won't. Please use another site.

**Reply by JanCourtheyn** [reply]
Via Google Drive now: https://drive.google.com/file/d/1LV-grzeeTL1ShL3P9efj4gZNaW84lFn9/view?usp=sharing

**Comment by snibgo** [comment]
Your PSD is CMYK, with embedded profile. So I use "-profile sRGB.icc" to convert it to sRGB.

Windows BAT syntax:
```
magick 46631035.psd[4] ^
  -profile sRGB.icc ^
  -resize 290x -rotate 90 +repage ^
  -gravity center -background None -extent 200x300 ^
  -channel RGB -threshold 15%% +channel ^
  -colorspace gray ^
  ( +clone ^
    -alpha extract -edge 5 ^
    -negate ^
  ) ^
  -compose Darken -composite ^
  -density 200x200 ^
  out.png
```
![out](https://user-images.githubusercontent.com/33812368/115741585-ccfd5900-a387-11eb-9148-c5e141c88f42.png)
As my processing is in sRGB space, not CMYK, the threshold should be adjusted to suit.

**Reply by JanCourtheyn** [reply]
Thanks a lot! This is awesome, much appreciated.
